docs/reference/modules/node.asciidoc

```diff
@@ -1,32 +1,48 @@
 [[modules-node]]
 == Node
 
-*elasticsearch* allows to configure a node to either be allowed to store
-data locally or not. Storing data locally basically means that shards of
-different indices are allowed to be allocated on that node. By default,
-each node is considered to be a data node, and it can be turned off by
-setting `node.data` to `false`.
+*elasticsearch* allows the configuration of a node to play multiple
+roles. By default, a node's role is a _data node_, a _master node_
+and a _client node_.
 
-This is a powerful setting allowing to simply create smart load
-balancers that take part in some of different API processing. Lets take
-an example:
+*data nodes* store indexed data locally and make it available for
+further manipulation. Storing data locally basically means that
+shards of different indices are allowed to be allocated on that
+node. The default is `node.data: true`, but can also be set to
+`node.data: false` in the configuration file.
 
-We can start a whole cluster of data nodes which do not even start an
-HTTP transport by setting `http.enabled` to `false`. Such nodes will
-communicate with one another using the
-<<modules-transport,transport>> module. In front
-of the cluster we can start one or more "non data" nodes which will
-start with HTTP enabled. All HTTP communication will be performed
-through these "non data" nodes.
+*master nodes* control cluster state, including changes such as node
```

Comment:
*master nodes* participate in master elections and, when elected as the _active master node_, control cluster state. If you choose to dedicate a node to just play the *master node* role then never send requests to it. This prevents the node from getting overloaded and lets it spend 100% of its time keeping cluster state up to date. The default...